src/main/java/bio/overture/rollcall/config/RollcallConfig.java

```diff
@@ -50,10 +51,12 @@ public void setAliases(@NonNull List<ConfiguredAlias> aliases) {
   @Data
   @NoArgsConstructor
   @AllArgsConstructor
+  @RequiredArgsConstructor
   public static class ConfiguredAlias {
-    private String alias;
-    private String entity;
-    private String type;
+    @NonNull private String alias;
+    @NonNull private String entity;
+    @NonNull private String type;
+    private int recentShardsToKeepOnRelease = -1;
```

Comment:
isn't this actually recent releases for a shard?

I kinda overloaded the term shard when I originally designed the grammar. Maybe we can call this something like releaseRotation or something like that similar to the concept of log rotation.

Reply:
Yea I struggled with trying to come up with a good name for this variable. I hadn't considered that but I agree `releaseRotation` does represent this behaviour better. Will update :+1: